nautobot/docs/models/extras/role.md

```diff
@@ -0,0 +1,81 @@
+# Roles
+
+The Role model represents a role that can be assigned to a device, rack, virtual machine, IP address, VLAN, or prefix. Each role is identified by a unique name and has a slug, weight, color and content-types associated with it.
+
+## Role Basics
+
+The value of a `role` field on a model (such as `Device.role`) will be represented as a `nautobot.extras.models.Role` object.
+
+When created, a `Role` can be associated to one or more model content-types using a many-to-many relationship. The relationship to each model is referenced across all user interfaces using the `{app_label}.{model}` naming convention (e.g. `dcim.device`).
+
+Roles may be managed by navigating to **Organization > Roles** in the navigation menu.
+
+### Importing Objects with a `role` Field
+
+When using CSV import to reference a `role` field on an object, the `Role.name` field is used.
+
+## Role Internals
+
+!!! warning
+    The section below is largely intended for developers who may need to create
+    data models of their own that implement a `role` field. Proceed at your
+    own risk!
+
+Any model that is intended to have a `role` field must inherit from either of these two mixins: `nautobot.extras.models.roles.RoleModelMixin` or `nautobot.extras.models.roles.RoleRequiredRoleModelMixin`. The `RoleModelMixin` adds a nullable `role` field, while the `RoleRequiredRoleModelMixin` adds a required `role` field. This abstract model will add an `nautobot.extras.models.roles.RoleField` to the model. The abstract base will automatically assign a `related_name` for the reverse relationship back to the inheriting model's name (e.g. `dcim_device_related`).
+
+### `RoleField` model field
+
+The `RoleField` field type subclass of a `django.db.models.ForeignKey`
+
+This model field also emits its own form field to eliminate the requirement for a form field to be explicitly added to model forms.
+
+### `RoleFilter` filter field
+
+Any filter that is intended to have a `role` field must inherit from `nautobot.extras.filters.RoleModelFilterSetMixin`. This will add a `nautobot.extras.filters.RoleFilter` to the filter, which allows filtering by the `name` of the role.
+
+### Form fields
+
+Any model form that is intended to have a `role` field must inherit from one of two mixins, depending on the use-case:
```

Comment:
This is misleadingly phrased. Not all model forms need these mixins (a basic create/edit form does not, for one). Should probably rephrase to make it clear that filter forms (only) need to use `RoleModelFilterFormMixin` and bulk-edit forms (only) need to use `RoleModelBulkEditFormMixin`.